js/src/components/contact-information/phone-number-card/verification-code-control.test.js

```diff
@@ -0,0 +1,110 @@
+ */
+import VerificationCodeControl from './verification-code-control';
+
+describe( 'VerificationCodeControl component', () => {
+	test( 'digit inputs are visible and empty', () => {
+		render( <VerificationCodeControl onCodeChange={ () => {} } /> );
+
+		const inputs = screen.getAllByRole( 'textbox' );
+		expect( inputs?.length ).toBe( 6 );
+		inputs.forEach( ( { value } ) => {
+			expect( value ).toBe( '' );
+		} );
+	} );
+
+	test( 'digit typing applies to the input', async () => {
+		render( <VerificationCodeControl onCodeChange={ () => {} } /> );
+		const input = screen.getAllByRole( 'textbox' )[ 0 ];
+
+		await userEvent.type( input, '2' );
+		expect( input.value ).toBe( '2' );
+
+		await userEvent.clear( input );
+		expect( input.value ).toBe( '' );
+
+		await userEvent.type( input, '5' );
+		expect( input.value ).toBe( '5' );
+
+		fireEvent.input( input, {
+			target: { value: '3' },
+		} );
+		expect( input.value ).toBe( '3' );
+	} );
+
+	test( 'clears if a non digit is typed', async () => {
+		render( <VerificationCodeControl onCodeChange={ () => {} } /> );
+		const input = screen.getAllByRole( 'textbox' )[ 0 ];
+
+		await userEvent.type( input, '2' );
+
+		fireEvent.input( input, {
+			target: { value: 'x' },
+		} );
+		expect( input.value ).toBe( '' );
+	} );
+
+	test( 'moves the focus with cursor keys', async () => {
+		render( <VerificationCodeControl onCodeChange={ () => {} } /> );
+		const inputs = screen.getAllByRole( 'textbox' );
+
+		await userEvent.type( inputs[ 0 ], '{arrowright}' );
+		expect( inputs[ 1 ] ).toHaveFocus();
+
+		await userEvent.type( inputs[ 1 ], '{arrowleft}' );
+		expect( inputs[ 0 ] ).toHaveFocus();
+	} );
+
+	test( 'moves the focus with typing and backspace', async () => {
+		render( <VerificationCodeControl onCodeChange={ () => {} } /> );
+		const inputs = screen.getAllByRole( 'textbox' );
+
+		await userEvent.type( inputs[ 0 ], '2' );
+		expect( inputs[ 1 ] ).toHaveFocus();
+
+		await userEvent.type( inputs[ 1 ], '{backspace}' );
+		expect( inputs[ 0 ] ).toHaveFocus();
+	} );
+
+	test( "last input doesn't move the cursor forwards", async () => {
+		render( <VerificationCodeControl onCodeChange={ () => {} } /> );
+		const inputs = screen.getAllByRole( 'textbox' );
+
+		await userEvent.type( inputs[ inputs.length - 1 ], '{arrowright}' );
+		expect( inputs[ inputs.length - 1 ] ).toHaveFocus();
+
+		await userEvent.type( inputs[ inputs.length - 1 ], '2' );
+		expect( inputs[ inputs.length - 1 ] ).toHaveFocus();
+	} );
+
+	test( "first input doesn't move the cursor backwards", async () => {
+		render( <VerificationCodeControl onCodeChange={ () => {} } /> );
+		const inputs = screen.getAllByRole( 'textbox' );
+
+		await userEvent.type( inputs[ 0 ], '{arrowleft}' );
+		expect( inputs[ 0 ] ).toHaveFocus();
+
+		await userEvent.type( inputs[ 0 ], '{backspace}' );
+		expect( inputs[ 0 ] ).toHaveFocus();
+	} );
+
+	test( 'typing calls onChange callback', async () => {
+		const onChange = jest.fn();
```

Comment:
:nail_care: Personally, I like to use named spies, to make nicer failure messages.

```suggestion
		const onChange = jest.fn().mockName('onChange callback');
```

Reply:
Nice, didn't know that one!